src/main/java/org/apache/maven/shared/archiver/PomPropertiesUtil.java

```diff
@@ -52,92 +48,53 @@ private Properties loadPropertiesFile(Path file) throws IOException {
         }
     }
 
-    private boolean sameContents(Properties props, Path file) throws IOException {
-        if (!Files.isRegularFile(file)) {
-            return false;
-        }
-
-        Properties fileProps = loadPropertiesFile(file);
-        return fileProps.equals(props);
-    }
-
-    private void createPropertiesFile(Properties properties, Path outputFile, boolean forceCreation)
-            throws IOException {
+    private void createPropertiesFile(Properties properties, Path outputFile) throws IOException {
         Path outputDir = outputFile.getParent();
-        if (outputDir != null && !Files.isDirectory(outputDir)) {
+        if (outputDir != null) {
             Files.createDirectories(outputDir);
         }
-        if (!forceCreation && sameContents(properties, outputFile)) {
-            return;
-        }
-
-        try (PrintWriter pw = new PrintWriter(outputFile.toFile(), StandardCharsets.ISO_8859_1.name());
-                StringWriter sw = new StringWriter()) {
-
-            properties.store(sw, null);
-
-            List<String> lines = new ArrayList<>();
-            try (BufferedReader r = new BufferedReader(new StringReader(sw.toString()))) {
-                String line;
-                while ((line = r.readLine()) != null) {
-                    if (!line.startsWith("#")) {
-                        lines.add(line);
-                    }
-                }
-            }
-
-            Collections.sort(lines);
-            for (String l : lines) {
-                pw.println(l);
-            }
+        // For reproducible builds, sort the properties and drop comments.
+        // The java.util.Properties class doesn't guarantee order so we have
+        // to write the file using a Writer.
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        properties.store(baos, null);
+        String nl = System.lineSeparator();
+        String output = baos.toString(StandardCharsets.ISO_8859_1)
```

Comment:
We're leaning pretty heavily on Hyrum's Law here. I'm not convinced we can rely on the description of current encoding in the Javadoc to apply to future encoding. In particular,  I would not be at all surprised to see properties.store write in UTF-8 in a future JDK.

Reply:
It's been this way since JDK 1.2, so 26 years.  And for the encoding, the Properties class has been opened to other encodings in JDK 1.5.  The default encoding for reading properties [has already been changed in JDK 9](https://docs.oracle.com/javase/9/intl/internationalization-enhancements-jdk-9.htm#GUID-974CF488-23E8-4963-A322-82006A7A14C7).   The encoding used here is kind irrelevant because all non-ascii chars have been transformed into `\uxxxx` sequences, thereby removing everything that is encoding specific.